### PR TITLE
fix: remove any name field restriction

### PR DESCRIPTION
**Pull request by obdulia-losantos**
Closes: https://jira.ehealthafrica.org/browse/AET-495

In kernel:
- Remove in `name` field the unique constraint for all models except for `Schema` model.
- Remove in `name` field the length size restriction in all models
- Remove in `type` field the length size restriction in `Schema` model
- Adapt `upsert_project_artefacts` method to new `name` restrictions.

In UI:
- Remove in `name` field the length size restriction

### CODE REVIEW

**Review by lordmallam — DISMISSED** [review]
LGMT, We will have to refactor the `ui - preflight` functions to remove the constraints as well

**Comment by obdulia-losantos** [comment]
There are no more "name" constraints in the preflight methods :wink:

File: aether-kernel/aether/kernel/api/models.py [hunk 1]
```diff
@@ -304,7 +304,7 @@ class Attachment(ExportModelOperationsMixin('kernel_attachment'), ProjectChildAb
 
     # http://www.linfo.org/file_name.html
     # Modern Unix-like systems support long file names, usually up to 255 bytes in length.
```
Comment (admbtlr) [inline aether-kernel/aether/kernel/api/models.py]: So we should probably take out this comment too, shouldn't we?

**Review by admbtlr — CHANGES_REQUESTED** [review]

File: aether-kernel/aether/kernel/api/models.py (hunk 1 above)
Comment (obdulia-losantos) [reply aether-kernel/aether/kernel/api/models.py]: Right!

**Review by admbtlr — APPROVED** [review]

**Pull request merged**